redhat-appstudio/service-provider-integration-operator
# feat: grafana panel for github error metrics

### What does this PR do?
$TITLE

### Screenshot/screencast of this PR

#### Panel itself:
![grafan_panel](https://user-images.githubusercontent.com/73115616/226291838-fb4e49b6-9087-4a62-b94b-8057622c8001.png)

#### Dashboard variant 1: In this variant, I just added the panel into the free space present in the dashboard.
![grafana_dashboard](https://user-images.githubusercontent.com/73115616/226291845-fac8e0c8-1b4f-4885-9418-b01c92cfe5ec.png)

#### Dashboard variant 2: In this variant, I reorder the panels so that GitHub panels are on the same x-axis.
![grafana_dashboard2](https://user-images.githubusercontent.com/73115616/226291722-b510a9e0-4ebb-4f65-9283-9ae6a562c755.png)

#### Dashboard variant 3: Similar to 2
![image](https://user-images.githubusercontent.com/73115616/226339073-f52b140b-cafe-4dab-9e28-668d3dae728d.png)



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-423

### How to test this PR?

Pick a variant you like more.

To deploy the dashboard yourself, follow these steps:
1. Replace `config/monitoring/base/grafana-dashboards/spi-outbound-traffic.json` with the content of this PR.
2. Deploy SPI (`make deploy_...`) and monitoring infrastructure (`hack/deploy-minikube-monitoring.sh`)
3. a) To see some graphs in the panel, you can replace SPI operator image: `quay.io/pbaran/spio:testing`. 
b) (Optional) So that the authorized calls are triggered, you should upload a valid GitHub token.
c) Run this script so that metric counters are increased: 

```shell
#!/bin/bash

for i in {1..20} ; do
    kubectl apply -f samples/spiaccesscheck.yaml
    sleep 6
    kubectl delete -f samples/spiaccesscheck.yaml
done
```

```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
